File: app/src/main/java/com/chooloo/www/callmanager/adapter/MenuAdapter.java
package com.chooloo.www.callmanager.adapter;

import android.content.Context;
import android.view.Menu;
import android.view.MenuItem;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.chooloo.www.callmanager.ui.listitem.ListItem;
import com.chooloo.www.callmanager.ui.listitem.ListItemHolder;

public class MenuAdapter extends RecyclerView.Adapter<ListItemHolder> {

    private final Menu mMenu;
    private final Context mContext;
    private OnMenuItemClickListener mOnMenuItemClickListener;

    public MenuAdapter(Context context, Menu menu) {
        mContext = context;
        mMenu = menu;
        mOnMenuItemClickListener = menuItem -> {
        };
    }

    @NonNull
    @Override
    public ListItemHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ListItemHolder(mContext);
    }

    @Override
    public void onBindViewHolder(@NonNull ListItemHolder holder, int position) {
        MenuItem menuItem = mMenu.getItem(position);
        ListItem listItem = holder.getListItem();

        listItem.setOnClickListener(view -> mOnMenuItemClickListener.onMenuItemClick(menuItem));
        listItem.setBigText((String) menuItem.getTitle());
        listItem.setImageDrawable(menuItem.getIcon());
    }

    @Override
    public int getItemCount() {
        return mMenu.size();
    }

    public void setOnMenuItemClickListener(OnMenuItemClickListener onMenuItemClickListener) {
        mOnMenuItemClickListener = onMenuItemClickListener;
    }

    public interface OnMenuItemClickListener {
        void onMenuItemClick(MenuItem menuItem);
    }
}

File: app/src/main/java/com/chooloo/www/callmanager/adapter/MenuAdapter.kt
package com.chooloo.www.callmanager.adapter

import android.content.Context
import android.view.Menu
import android.view.MenuItem
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.chooloo.www.callmanager.ui.widgets.ListItemHolder

class MenuAdapter(
        private val _context: Context,
        private val _menu: Menu
) : RecyclerView.Adapter<ListItemHolder>() {
    private var _onMenuItemClickListener: ((MenuItem) -> Unit?)? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListItemHolder {
        return ListItemHolder(_context)
    }

    override fun onBindViewHolder(holder: ListItemHolder, position: Int) {
        val menuItem = _menu.getItem(position)
        holder.listItem.apply {
            setOnClickListener { _onMenuItemClickListener?.invoke(menuItem) }
            setBigText(menuItem.title as String)
            setImageDrawable(menuItem.icon)
        }
    }

    override fun getItemCount(): Int {
        return _menu.size()
    }

    fun setOnMenuItemClickListener(onMenuItemClickListener: ((MenuItem) -> Unit?)?) {
        _onMenuItemClickListener = onMenuItemClickListener
    }
}